gcp/config.go

```diff
@@ -0,0 +1,240 @@
+package gcp
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"strings"
+)
+
+const (
+	DefaultZone        = "us-central1-f"
+	DefaultMachineType = "f1-micro"
+	BaseCalyptiaImage  = "projects/calyptia-infra/global/images/family/gold-calyptia-core"
+)
+
+// metadata
+type metadata struct {
+	Items []item `yaml:"items"`
+}
+
+// Items
+type item struct {
+	Key   string `yaml:"key"`
+	Value string `yaml:"value"`
+}
+
+// tags
+type tags struct {
+	Items []string `yaml:"items"`
+}
+
+// disk
+type disk struct {
+	DeviceName       string           `yaml:"deviceName"`
+	Type             string           `yaml:"type"`
+	Boot             bool             `yaml:"boot"`
+	AutoDelete       bool             `yaml:"autoDelete"`
+	InitializeParams initializeParams `yaml:"initializeParams"`
+}
+
+// networkInterface
+type networkInterface struct {
+	Network       string         `yaml:"network"`
+	AccessConfigs []accessConfig `yaml:"accessConfigs"`
+}
+
+// accessConfig
+type accessConfig struct {
+	Name  string `yaml:"name"`
+	NatIP string `yaml:"natIP"`
+}
+
+// Config
+type Config struct {
+	projectID string      `yaml:"-"`
+	Resources []resources `yaml:"resources"`
+	Outputs   []output    `yaml:"outputs"`
+}
+
+// resources
+type resources struct {
+	Type       string     `yaml:"type"`
+	Name       string     `yaml:"name"`
+	Properties properties `yaml:"properties"`
+}
+
+// properties
+type properties struct {
+	Zone              string             `yaml:"zone"`
+	MachineType       string             `yaml:"machineType"`
+	Metadata          metadata           `yaml:"metadata"`
+	Tags              tags               `yaml:"tags"`
+	Disks             []disk             `yaml:"disks"`
+	NetworkInterfaces []networkInterface `yaml:"networkInterfaces"`
+}
+
+// initializeParams
+type initializeParams struct {
+	SourceImage string `yaml:"sourceImage"`
+}
+
+// output
+type output struct {
+	Name  string `yaml:"name"`
+	Value string `yaml:"value"`
+}
+
+func NewConfig(projectID string) Config {
+	c := Config{
+		projectID: projectID,
+		Resources: []resources{
+			{
+				Type: "compute.v1.instance",
+				Name: "calyptia-core-instance",
+				Properties: properties{
+					Zone:        DefaultZone,
+					MachineType: fmt.Sprintf("projects/%s/zones/%s/machineTypes/%s", projectID, DefaultZone, DefaultMachineType),
+					Metadata:    metadata{Items: []item{}},
+					Tags:        tags{Items: []string{"calyptia-core-instance"}},
```

Comment:
Tags has to be extended with 2 things: 1 ) instance name 2) environment name, so you can filter out later when deleting.